packages/passport-interface/src/SubscriptionManager.ts

```diff
@@ -123,14 +139,15 @@ export class FeedSubscriptionManager {
    * repopulated, so callers should check this in order to determine success.
    */
   private async fetchSingleSubscription(
-    subscription: Subscription
+    subscription: Subscription,
+    identity: Identity
   ): Promise<SubscriptionActions[]> {
     const responses: SubscriptionActions[] = [];
     this.resetError(subscription.id);
     try {
       const result = await this.api.pollFeed(subscription.providerUrl, {
         feedId: subscription.feed.id,
-        pcd: subscription.credential
+        pcd: await this.generateCredential(subscription, identity)
```

Comment:
again - isn't this preventing us from supporting feeds that have another credential type?